Automattic/woocommerce-payments
# Payouts: Add payout bank reference key to payout reports and CSV

Fixes #5316 

>[!NOTE]
>  Note added after PR approval - for future reference - `Trace ID` was renamed to `Bank reference key`, based on [review feedback](https://github.com/Automattic/transact-platform-server/pull/6917#pullrequestreview-2468911647) and p1732861507747189/1732857021.096589-slack-C02BW3Z8SHK. If you spot trace ID anywhere in the discussion below ( other than Stripe Payout object ), please read it as `Bank reference key`

#### Changes proposed in this Pull Request

* Add `Trace ID` column in payouts list seen within `Payments > Payouts` , and within CSV exports. The `Trace ID` will help merchants match inbound payout entries in their bank statements with WooPayments Payouts. 
* This will be visible for all Payouts after November 21, 2024

#### Testing instructions

* Please check out your local server to the branch - `add/6838-payout-trace-id` , unless it is already merged. 
* Do a `npm run down && npm run up` on the server to restart docker and make sure the `trace_id` column is added to the payouts cache tables. 
* Check in to this branch on the client
* Make sure your linked account on the client has a few payouts after November 21, 2024.
* You may link to `acct_1OuYn1QpUShfetUA` if you like, since that account has a few sample payouts that can be helpful for testing. 
* Browse to `Payments > Deposits` 
* You should see the `Trace ID` column and the ID on the payouts after November 21, 2024.

<img src="https://github.com/user-attachments/assets/d6473339-37cc-4ed2-8d22-8ff6f5cce53d" width="500px" /> 

##### Test browser export
* On the Payouts report page, use a filter to have payouts less than 25 and click on download
* Download the export file, and check the Trace ID's appearing in the report

##### Test endpoint export
* Make sure you have more than 25 payouts in your account
* On the Payouts report page, click on the download button.
* You should see a message that the report would be emailed to you
* Login to the wp-admin of the local server
* Check the email log and click the download link in the email you see there for the export
* You should see the Trace ID column with the data correctly in the CSV file

-------------------


**Post merge**
